src/main/java/com/internship/socialnetwork/service/impl/PostServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.internship.socialnetwork.service.impl;
+
+import com.internship.socialnetwork.dto.PostDto;
+import com.internship.socialnetwork.exception.NotFoundException;
+import com.internship.socialnetwork.model.Post;
+import com.internship.socialnetwork.repository.PostRepository;
+import com.internship.socialnetwork.service.PostService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class PostServiceImpl implements PostService {
+
+    private final PostRepository postRepository;
+
+    @Override
+    public Post getModel(Long id) {
+        return postRepository.findById(id).orElseThrow(() -> new NotFoundException("Post with id: " + id + " not found!"));
```

Comment:
This string can be a constant, you can use String.format for interpolating id value